diff --git a/master/master.c b/master/master.c
index 6c724478db..781eb8e8ab 100644
--- a/master/master.c
+++ b/master/master.c
@@ -2198,7 +2198,9 @@ int main(int argc, char **argv)
             int exit_result = EX_OSERR;
 
             /* Tell our parent that we failed. */
-            r = write(startup_pipe[1], &exit_result, sizeof(exit_result));
+            if (write(startup_pipe[1], &exit_result, sizeof(exit_result)) == -1) {
+                syslog(LOG_ERR, "can't write to startup parent pipe: %m");
+            }
 
             fatal("setsid failure", EX_OSERR);
         }
@@ -2211,10 +2213,13 @@ int main(int argc, char **argv)
     if(pidfd == -1) {
         int exit_result = EX_OSERR;
 
+        syslog(LOG_ERR, "can't open pidfile: %m");
+
         /* Tell our parent that we failed. */
-        r = write(startup_pipe[1], &exit_result, sizeof(exit_result));
+        if (daemon_mode && write(startup_pipe[1], &exit_result, sizeof(exit_result)) == -1) {
+            syslog(LOG_ERR, "can't write to startup parent pipe: %m");
+        }
 
-        syslog(LOG_ERR, "can't open pidfile: %m");
         exit(EX_OSERR);
     } else {
         char buf[100];
@@ -2223,7 +2228,9 @@ int main(int argc, char **argv)
             int exit_result = EX_OSERR;
 
             /* Tell our parent that we failed. */
-            r = write(startup_pipe[1], &exit_result, sizeof(exit_result));
+            if (write(startup_pipe[1], &exit_result, sizeof(exit_result)) == -1) {
+                syslog(LOG_ERR, "can't write to startup parent pipe: %m");
+            }
 
             fatal("cannot get exclusive lock on pidfile (is another master still running?)", EX_OSERR);
         } else {
@@ -2234,10 +2241,14 @@ int main(int argc, char **argv)
             if (pidfd_flags == -1) {
                 int exit_result = EX_OSERR;
 
+                syslog(LOG_ERR, "unable to set close-on-exec for pidfile: %m");
+
                 /* Tell our parent that we failed. */
-                r = write(startup_pipe[1], &exit_result, sizeof(exit_result));
+                if (write(startup_pipe[1], &exit_result, sizeof(exit_result)) == -1) {
+                    syslog(LOG_ERR, "can't write to startup parent pipe: %m");
+                }
 
-                fatalf(EX_OSERR, "unable to set close-on-exec for pidfile: %m");
+                fatalf(EX_OSERR, "unable to set close-on-exec for pidfile (see syslog for details)");
             }
 
             /* Write PID */
@@ -2247,10 +2258,14 @@ int main(int argc, char **argv)
                write(pidfd, buf, strlen(buf)) == -1) {
                 int exit_result = EX_OSERR;
 
+                syslog(LOG_ERR, "unable to write to pidfile: %m");
+
                 /* Tell our parent that we failed. */
-                r = write(startup_pipe[1], &exit_result, sizeof(exit_result));
+                if (daemon_mode && write(startup_pipe[1], &exit_result, sizeof(exit_result)) == -1) {
+                    syslog(LOG_ERR, "can't write to startup parent pipe: %m");
+                }
 
-                fatalf(EX_OSERR, "unable to write to pidfile: %m");
+                fatalf(EX_OSERR, "unable to write to pidfile (see syslog for details)");
             }
             if (fsync(pidfd))
                 fatalf(EX_OSERR, "unable to sync pidfile: %m");
