diff --git a/Dockerfile b/Dockerfile
index 50fc755..e734871 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -2,8 +2,16 @@ FROM ubuntu:14.04
 
 RUN apt-get update && \
     apt-get install -y apache2 php5 php5-sqlite php5-curl && \
-    apt-get clean && rm -rf /var/lib/apt/lists/* && \
-    echo "ServerName localhost" >> /etc/apache2/apache2.conf
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+RUN echo \
+    "ServerName localhost\n" \
+    "<Directory /var/www/html/data/>\n" \
+    "    Deny from all\n" \
+    "</Directory>\n" \
+    "<Directory /var/www/html/data/favicons/>\n" \
+    "    Allow from all\n" \
+    "</Directory>\n" >> /etc/apache2/apache2.conf
 
 COPY . /var/www/html
 
