From dd4d683f1248d50d1bd95e64c8c69088ecb13d2e Mon Sep 17 00:00:00 2001
From: Conrad Kleinespel <conradk@conradk.com>
Date: Mon, 29 Jun 2015 13:04:18 +0200
Subject: [PATCH] adds syslogging when RAWX is unavailable

---
 tools/rawx-monitor.py | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/tools/rawx-monitor.py b/tools/rawx-monitor.py
index 3817ed2728..fd968844c3 100755
--- a/tools/rawx-monitor.py
+++ b/tools/rawx-monitor.py
@@ -19,6 +19,7 @@
 
 import sys
 import urllib2
+import syslog
 
 RAWX_STAT_KEYS = [
 	("rawx.reqpersec",	"stat.total_reqpersec"),
@@ -49,12 +50,16 @@ def parse_info(stream):
 	return data
 
 def get_stat_lines(url, stat_keys):
-	stream = urllib2.urlopen(url)
-	data = parse_info(stream)
-	stream.close()
-	stats = [("%s = %s" % (k[1], str(data[k[0]])))
-			for k in stat_keys if k[0] in data]
-	return stats
+	try:
+		stream = urllib2.urlopen(url)
+		data = parse_info(stream)
+		stream.close()
+		stats = [("%s = %s" % (k[1], str(data[k[0]])))
+				for k in stat_keys if k[0] in data]
+		return stats
+	except urllib2.URLError as e:
+		syslog.syslog(syslog.LOG_ERR, "rawx-monitor could not connect to RAWX server at %s: %s" % (url, e.strerror))
+		sys.exit(1)
 
 def main(args):
 	ip_port = str(args[1]).split("|")[2]
