diff --git a/tools/sds-bootstrap.py b/tools/sds-bootstrap.py
index b1d45a3e10..1a650a2550 100755
--- a/tools/sds-bootstrap.py
+++ b/tools/sds-bootstrap.py
@@ -17,7 +17,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 from string import Template
-import os, errno
+import os, errno, pwd
 
 #env.G_DEBUG=fatal_warnings
 #env.G_SLICE=debug-blocks
@@ -517,7 +517,7 @@ def getint(v,default):
 			SDSDIR=SDSDIR, TMPDIR=TMPDIR,
 			DATADIR=DATADIR, CFGDIR=CFGDIR, RUNDIR=RUNDIR, SPOOLDIR=SPOOLDIR,
 			LOGDIR=LOGDIR, CODEDIR=CODEDIR,
-			UID=str(os.geteuid()), GID=str(os.getgid()), USER=str(os.getlogin()),
+			UID=str(os.geteuid()), GID=str(os.getgid()), USER=str(pwd.getpwuid(os.getuid())),
 			VERSIONING=versioning, STGPOL=stgpol,
 			M2_REPLICAS=meta2_replicas, M2_DISTANCE=str(1),
 			SQLX_REPLICAS=sqlx_replicas, SQLX_DISTANCE=str(1))
