From 8da631a897822d3f55364368f10f9d340e2ff9fc Mon Sep 17 00:00:00 2001
From: Conrad Kleinespel <conradk@conradk.com>
Date: Sun, 13 Sep 2015 22:04:27 +0200
Subject: [PATCH] updates package version in docs

---
 src/lib.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib.rs b/src/lib.rs
index 8b060b6daf..66bcd6b670 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -26,7 +26,7 @@
 //!
 //! ```toml
 //! [dependencies]
-//! regex = "0.1.8"
+//! regex = "0.1.41"
 //! ```
 //!
 //! and this to your crate root:
