From fc32db3ef56834ff713ecdd2f6b736292a8b1c0a Mon Sep 17 00:00:00 2001
From: Conrad Kleinespel <conradk@conradk.com>
Date: Wed, 29 Jun 2016 17:50:11 +0200
Subject: [PATCH] transmission: makes "from" address optional

---
 lib/SparkPost/Transmission.php | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/SparkPost/Transmission.php b/lib/SparkPost/Transmission.php
index 6c45c7e..4c2eda8 100644
--- a/lib/SparkPost/Transmission.php
+++ b/lib/SparkPost/Transmission.php
@@ -90,7 +90,9 @@ private function formatCarbonCopy($payload)
      */
     private function formatShorthandRecipients($payload)
     {
-        $payload['content']['from'] = $this->toAddressObject($payload['content']['from']);
+        if (isset($payload['content']['from'])) {
+            $payload['content']['from'] = $this->toAddressObject($payload['content']['from']);
+        }
 
         for ($i = 0; $i < count($payload['recipients']); ++$i) {
             $payload['recipients'][$i]['address'] = $this->toAddressObject($payload['recipients'][$i]['address']);
